### PR TITLE
Add 'che-theia-terminal-extension' and 'che-theia-task-plugin' to the Theia image

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Add 'che-theia-terminal-extension', 'che-theia-task-plugin' to the Theia image. 
Integration che-machine-exec.
Remove default Theia terminal extension from package.json of the Theia image.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10314

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 1]
```diff
@@ -45,12 +44,24 @@
...
+              },
+              "installers": []
             }
           },
           "recipe": {
-            "contentType": "",
-            "type": "dockerimage",
-            "content": "eclipse/che-theia:0.3.14-nightly"
+            "contentType": "application/x-yaml",
+            "type": "compose",
```
Comment (garagatyi) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json:63]: FYI compose is not supported on openshift and kubernetes infras

Comment (AndrienkoAleksandr) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json:63]: @garagatyi I changed two stacks. First one for docker. Second one for openshift

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 2]
```diff
@@ -45,12 +44,24 @@
...
+            "content": "services:\n  machine-exec:\n    image: 'aandrienko/che-machine-exec:latest'\n    mem_limit: 536870912\n  theia:\n    image: 'eclipse/che-theia:0.3.14-nightly'\n    mem_limit: 1073741824\n    depends_on:\n      - machine-exec\n"
```
Comment (riuvshin) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json]: is that `image: 'aandrienko/che-machine-exec:latest` right image to have in master?

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 3]
```diff
@@ -119,11 +130,23 @@
...
+                  "port": "4444"
+                }
+              },
+              "installers": []
             }
           },
           "recipe": {
             "type": "openshift",
-            "content": "---\nkind: List\nitems:\n-\n  apiVersion: v1\n  kind: Pod\n  metadata:\n    name: ws\n  spec:\n    containers:\n      -\n        image: eclipse/che-theia:0.3.14-nightly\n        name: theia\n      -\n        image: eclipse/che-dev:nightly\n        name: dev",
+            "content": "---\nkind: List\nitems:\n-\n  apiVersion: v1\n  kind: Pod\n  metadata:\n    name: ws\n  spec:\n    containers:\n      -\n        image: eclipse/che-theia:0.3.14-nightly\n        name: theia\n      -\n        image: eclipse/che-dev:nightly\n        name: dev\n      -\n        image: aandrienko/che-machine-exec:latest\n        name: machine-exec",
```
Comment (riuvshin) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json]: same question

File: ide/che-core-ide-stacks/src/main/resources/stacks.json (hunk 2 above)
Comment (AndrienkoAleksandr) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json]: Yes, exactly. CQ for kubernetes lib for che-machine-exec spends a lot of time, for now we can't merge it, so I used image from my fork.

Comment (riuvshin) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json]: cc: @skabashnyuk

**Review by ghost — APPROVED** [review]

File: dockerfiles/theia/src/extensions.json [hunk 4]
```diff
@@ -47,6 +47,20 @@
             "folder": "che-theia-java-extension",
             "checkoutTo": "master",
             "type": "git"
+        },
+        {
+            "name": "che-theia-terminal",
+            "source": "https://github.com/eclipse/che-theia-terminal-extension.git",
+            "folder": "che-theia-terminal",
+            "checkoutTo": "ee583bf4e7956502a2d9a4fab55b06bcbe7b0117",
```
Comment (benoitf) [inline dockerfiles/theia/src/extensions.json:55]: should we use a tag instead ?

File: dockerfiles/theia/src/extensions.json [hunk 5]
```diff
@@ -47,6 +47,20 @@
...
+            "type": "git"
+        },
+        {
+            "name": "@eclipse-che/theia-task-extension",
+            "source": "https://github.com/eclipse/che-theia-task-plugin.git",
+            "folder": "che-theia-task-extension",
+            "checkoutTo": "47450ee167a7d8429550c45be48f93230bf44ebd",
```
Comment (benoitf) [inline dockerfiles/theia/src/extensions.json:62]: should we use a tag instead ?

File: ide/che-core-ide-stacks/src/main/resources/stacks.json (hunk 2 above)
Comment (skabashnyuk) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json]: ok

**Review by skabashnyuk — APPROVED** [review]

**Review by mmorhun — APPROVED** [review]

**Review by ashumilova — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

File: ide/che-core-ide-stacks/src/main/resources/stacks.json (hunk 1 above)
Comment (jonahgraham) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json:63]: This change broke running Theia on kubernetes, see #11260. There is now no longer a stack that runs on kubernetes.

**Pull request merged**